### PR TITLE
[PM-12747] Move CollectionService and models to AC Team

### PR DESCRIPTION
**Pull request by eliykat**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-12747
## 📔 Objective

Move `CollectionService` and associated models into `@bitwarden/admin-console/common`.

I also took the opportunity to rename the implementation to `DefaultCollectionService` per our current naming convention, and drop the `Abstraction` alias where it was used.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by djsmith85 — COMMENTED** [review]
Could you please add `@bitwarden/admin-console/common` as a dependency to the `package.json`-files for `libs/importer`, `libs/tools/vault-export-core` and `libs/tools/vault-export-ui`. This is currently not enforced nor used, but in preparation to move to something like https://nx.dev/ for building packages within our monorepo.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Auth changes look good!

File: libs/importer/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   },
   "dependencies": {
     "@bitwarden/common": "file:../common",
-    "@bitwarden/vault-export-core": "file:../tools/export/vault-export/vault-export-core"
+    "@bitwarden/vault-export-core": "file:../tools/export/vault-export/vault-export-core",
+    "@bitwarden/admin-console-common": "file:../admin-console/src/common"
```
Comment (djsmith85) [inline libs/importer/package.json:23]: This package-name differs from the actually used package-name (`@bitwarden/admin-console/common`). Same issue applies to the vault-export package.json files. The one used here is a valid package name.  I hadn't caught that with your previous PR. While working on the exporters I ran into a similar issue with `@bitwarden/vault-export/core` and `@bitwarden/vault-export/ui`. These are invalid package names.

`libs/admin-console` does not export the `common`-folder as it's own package. It's only achieved by deeplinking within the tsconfigs. Should `libs/admin-console/common`be a it's own package or should it just be exported via `@bitwarden/admin-console` ?

I'm still fine with approving this, to not block the ownership move, but would like to request a follow-up ticket to look into a naming and packaging scheme for `libs/admin-console`.

The tools-team has chosen to separate their packages by features so we can easily build what we need and also only depend on things we need. We are obviously fortunate that our ownership splits up so nicely, which isn't as easy for the other teams. Collections for example are a very commonly used entity and overlap into a ton of domains.

**Review by djsmith85 — APPROVED** [review]
I went ahead and fixed any issues with incorrect import-paths (admin-console/src/common)

**Review by Jingo88 — APPROVED** [review]

**Review by jrmccannon — DISMISSED** [review]

File: libs/importer/package.json (hunk 1 above)
Comment (eliykat) [reply libs/importer/package.json:23]: I'll hit you up on Slack to discuss this further!

**Review by JaredSnider-Bitwarden — DISMISSED** [review]

**Review by justindbaur — DISMISSED** [review]

**Review by djsmith85 — DISMISSED** [review]

**Review by JaredSnider-Bitwarden — DISMISSED** [review]

**Review by djsmith85 — DISMISSED** [review]

**Review by Jingo88 — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Review by Jingo88 — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Comment by addisonbeck** [comment]
🥳

**Pull request merged**